middleware/plugin/disabled.go

```diff
@@ -0,0 +1,25 @@
+//go:build !plugin
+
+package plugin
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"errors"
+	"net/http"
+)
+
+type DummyMW struct{}
+
+func (mw *DummyMW) Wrap(next http.Handler) http.Handler { return next }
+func (mw *DummyMW) Shutdown()                           {}
+
+func NewMiddleware(Plugin, chan error) (*DummyMW, error) {
+	return nil, errors.New("this version of wfx was compiled without support for plugins")
```

Comment:
Nitpick: it still is compiled w/o plugin support?

Reply:
I've refactored the code. Now it's no longer possible to specify the plugin flag if wfx was built without support for plugins.